packages/opentelemetry-exporter-collector-grpc/src/CollectorExporterNodeBase.ts

```diff
@@ -55,25 +55,17 @@ export abstract class CollectorExporterNodeBase<
     onSuccess: () => void,
     onError: (error: collectorTypes.CollectorExporterError) => void
   ): void {
-    const promise = new Promise<void>(resolve => {
-      const _onSuccess = (): void => {
-        onSuccess();
-        _onFinish();
-      };
-      const _onError = (error: collectorTypes.CollectorExporterError): void => {
-        onError(error);
-        _onFinish();
-      };
-      const _onFinish = () => {
-        resolve();
-        const index = this._sendingPromises.indexOf(promise);
-        this._sendingPromises.splice(index, 1);
-      };
-
-      this._send(this, objects, _onSuccess, _onError);
-    });
+    const promise = new Promise<void>((resolve, reject) => {
+      this._send(this, objects, resolve, reject);
+    })
+      .then(onSuccess)
+      .catch(onError);
```

Comment:
If the `onSuccess` callback throws, the `onError` callback will be called unexpectedly. Passing them both to `.then` will ensure `onError` is only called if the original promise throws.

```suggestion
      .then(onSuccess, onError);
```

Reply:
Nice catch